fuzzers/libafl-fuzz/src/corpus.rs

```diff
@@ -0,0 +1,189 @@
+    Ok(())
+}
+
+pub fn set_solution_filepath(
+    state: &mut LibaflFuzzState,
+    testcase: &mut Testcase<BytesInput>,
+    output_dir: &PathBuf,
+) -> Result<(), Error> {
+    // sig:0SIGNAL
+    // TODO: verify if 0 time if objective found during seed loading
+    let mut filename = generate_base_filename(state);
+    let mut dir = "crashes";
+    if testcase
+        .hit_objectives()
+        .contains(&Cow::Borrowed("TimeoutFeedback"))
+    {
+        filename = format!("{filename},+tout");
+        dir = "hangs";
+    }
+    *testcase.file_path_mut() = Some(output_dir.join(dir).join(&filename));
+    *testcase.filename_mut() = Some(filename);
+    Ok(())
+}
+
+fn parse_time_line(line: &str) -> Result<u64, Error> {
+    line.split(": ")
+        .last()
+        .ok_or(Error::illegal_state("invalid stats file"))?
+        .parse()
+        .map_err(|_| Error::illegal_state("invalid stats file"))
+}
+
+pub fn check_autoresume(
+    fuzzer_dir: &Path,
+    intial_inputs: &PathBuf,
+    auto_resume: bool,
+) -> Result<Flock<File>, Error> {
+    if !fuzzer_dir.exists() {
+        std::fs::create_dir(fuzzer_dir)?;
+    }
+    // lock the fuzzer dir
+    let fuzzer_dir_fd = File::open(fuzzer_dir)?;
+    let file = match Flock::lock(fuzzer_dir_fd, FlockArg::LockExclusiveNonblock) {
+        Ok(l) => l,
+        Err(err) => match err.1 {
+            Errno::EWOULDBLOCK => return Err(Error::illegal_state(
+                "Looks like the job output directory is being actively used by another instance",
+            )),
+            _ => {
+                return Err(Error::last_os_error(
+                    format!("Error creating lock for output dir: exit code {}", err.1).as_str(),
+                ))
+            }
+        },
+    };
+    // Check if we have an existing fuzzed fuzzer_dir
+    let stats_file = fuzzer_dir.join("fuzzer_stats");
+    if stats_file.exists() {
+        let file = File::open(&stats_file).unwrap();
+        let reader = BufReader::new(file);
+        let mut start_time: u64 = 0;
+        let mut last_update: u64 = 0;
+        for (index, line) in reader.lines().enumerate() {
+            match index {
+                // first line is start_time
+                0 => {
+                    start_time = parse_time_line(&line?).unwrap();
+                }
+                // second_line is last_update
+                1 => {
+                    last_update = parse_time_line(&line?).unwrap();
+                }
+                _ => break,
+            }
+        }
+        if !auto_resume && last_update.saturating_sub(start_time) > OUTPUT_GRACE * 60 {
+            return Err(Error::illegal_state("The job output directory already exists and contains results! use AFL_AUTORESUME=true or provide \"-\" for -i "));
+        }
+    }
+    if auto_resume {
+        // TODO: once the queue stuff is implemented finish the rest of the function
```

Comment:
instead of writing // TODO here, write `todo!(...)` to make sure it's not used by accident by anyone